### DISCUSSION TITLE
Using `diesel2` instead of `sqlx`, maybe under a feature

### DISCUSSION THREAD

**Discussion by fenollp** [maxcountryman/underway#2] [Ideas]
Hi! Just noticed this crate.
We're relying on `diesel` to connect to pg and I see this uses `sqlx`.
I'm not sure if/how the two can "live together". Maybe you have experience with this?

Would you be open to allow users of this crate to use diesel instead of sqlx? 
Thanks

**Comment by maxcountryman** [comment]
I think more of this implementation could be moved into PostgreSQL itself (this is what a lot of these Postgres queues tend to do, e.g. `pg_boss`). That would make it easier to abstract over `sqlx` and support other drivers (it'd be nice to also be able to `tokio-postgres` for example).
